Move console.log to console.error

Digi59404 opened the pull request:
Some CI Tools such as GitLab sanitize their log data to not show .log commands. Because of this, an error will not be output into the console. By using console.error - CI Tools such as GitLab will display there error. When using .log they will not.

Digi59404 committed:
Move console.log to console.error

Some CI Tools such as GitLab sanitize their log data to not show .log commands. Because of this, an error will not be output into the console. By using console.error - CI Tools such as GitLab will display there error. When using .log they will not.

frsechet commented on index.js:
```diff
@@ -32,17 +32,17 @@ client.getSecretValue({ SecretId }, (err, data) => {
       const keys = Object.keys(secrets);
       keys.forEach(k => {
-        console.log(`${k}=${secrets[k]}`);
+        console.error(`${k}=${secrets[k]}`);
```
should remain console.log

frsechet commented on index.js:
```diff
@@ -32,17 +32,17 @@ client.getSecretValue({ SecretId }, (err, data) => {
     }
     catch (e) {
-      console.log(data.SecretString);
+      console.error(data.SecretString);
```
should remain console.log

frsechet commented on index.js:
```diff
@@ -32,17 +32,17 @@ client.getSecretValue({ SecretId }, (err, data) => {
   else {
     let buff = new Buffer(data.SecretBinary, "base64");
-    console.log(buff.toString("ascii"));
+    console.error(buff.toString("ascii"));
```
should remain console.log

frsechet commented:
Gitlab absolutely does show .log (this is where I have been using this script for months on about 50 different projects, I would probably have noticed by now 😉). Do you have a link to gitlab's documentation for this?

However I do agree that the errors could be written to console.error for easier separation of error/regular logs. Can you update your PR as per the review?

I assume that you are coming from my [Medium post](https://medium.com/clevyio/managing-secrets-in-ci-stages-with-aws-secrets-manager-7240938ddf41). This is mostly a quick and dirty tool, I encourage you to fork and modify it to your actual needs! For example, you could also write to an actual file, instead of just console.log. Whatever works for you!

Digi59404 committed:
Make non-errors a log.

Digi59404 commented:
Hey @frsechet  - I've went ahead and moved the two non-errors back to a console.log.

For GitLab, the .error and .log reporting is dependent upon the runner and its environment. I work for GitLab in Professional Services; We were using your project to show a potential client an example of how it's done. In our runner environment (AWS => Docker Executor) the security permissions did not allow console.log to show the errors. As soon as I switched the errors from .log to .error they showed up as intended!

Now, when we run the pipeline if there's an error. It's output the console, and because we're using GitLab Runners in Kubernetes. Kubernetes takes the logs from the runner failure and saves them to a central aggregated logging system.

This may be a quick and dirty tool in your mind. But it absolutely does the job, and it does it well. I've advised our client during the demo of it so they may contribute back in the future. But over-all it's a solid tool. It's being used in production.

frsechet commented:
Thanks a lot for the background explanation 😃 
Very happy to know that it is useful!

New build will be available in a short while from docker hub. Thanks!
